### ISSUE TITLE
Support for C64, Atari, Amiga, etc?

### ISSUE THREAD

**Issue by Adem1987** [OtherCrashOverride/go-play#21]
Could you implement more emulators to play the games from these platforms?

**Comment by OtherCrashOverride** [comment]
This was discussed on the forums.  One of the challenges is what to do about a keyboard?

**Comment by DavidKnight247** [comment]
The Atari 2600 could be a promising candidate. Stella is a mature emulator.

**Comment by kamotswind** [comment]
It may be a good idea to poll the users on the forum and see what systems most people would like to see that could also run on the ESP32's CPU and fit in the limited RAM (which OtherCrashOverride just told me about). It seems there is only about 400kb of RAM for whatever emulator to run in.

**Comment by Adem1987** [comment]
Simulating the keyboard inputs, till we can play with the "joystick"?

**Comment by rickyzhang82** [comment]
@kamotswind RAM on ESP32 chip is limited to 400KiB. They are also fragmented. But there is 4MB external RAM through SPI. You can not do DMA through external RAM. That's what I learnt so far in frame buffer improvement.

**Comment by jpxdude** [comment]
I would love to see these emulated on Odroid-Go!  I think Atari through Stella may be the most feasible option.  C64 might need keyboard emulation, or some kind of command menu (I think VICE has this?) and Amiga emulation is supposed to be quite taxing.

**Comment by jpxdude** [comment]
Someone on the forum suggested Atari Lynx.  Is that possible?

**Comment by DavidKnight247** [comment]
Probably, I have ported Handy to another handheld in the past. I had frame
rate issues on a faster chip though and there are issues with I2C data
transfer speed so frameskip is inevitable.

**Comment by OtherCrashOverride** [comment]
ColecoVision was added in the latest release:
https://github.com/OtherCrashOverride/go-play/releases/tag/20180718-smsplusgx

**Comment by hpcodecraft** [comment]
I'd like to add NeoGeo Pocket and NeoGeo Pocket Color to the list. They had only four buttons so that would work with the Odroid GO. I don't know if emulators have been ported to the chipset though

**Comment by kamotswind** [comment]
NeoGeo may be fairly difficult since I think it uses a lot more RAM then what is available.

**Comment by hpcodecraft** [comment]
@kamotswind I meant the NeoGeo **Pocket**, not the home console. I just checked the specs, it had 12K of RAM so that could work. See here: https://en.wikipedia.org/wiki/Neo_Geo_Pocket_Color#Technical_specifications

**Comment by kamotswind** [comment]
@hpcodecraft I believe the emulator would need to allocate at least 12KB for the 900H, 4KB for the Z80, 64KB for ROM code (unless the ROM image is written to SRAM or something), and all of that is not so bad but then there may be up to 4096KB needed for the actual game that is running. Emulators usually need to load the ROM in to RAM I believe.

**Comment by jutleys** [comment]
How about CPS Amstrad, Game&watch, MSX, Vertex,

**Comment by JasonB32** [comment]
Yes please Atari 7800?? I can't even believe how awesome the 7800 games are, way better than 2600 no offense anyone.

**Comment by JasonB32** [comment]
Could we get MSX games, I just found out about them and their pretty awesome? thx!!!

**Comment by Luppie1975** [comment]
Please look into the option to add MSX. Preferably MSX2. Most games can be played without a keyboard, some games that needs a keyboard for player selection are patched, so a joystick can be used to start the game. On github look into Openmsx. This is the most accurate emulator for multiple platforms.

**Comment by OtherCrashOverride** [comment]
I attempted an Atari Lynx port, but was not successful.
https://forum.odroid.com/viewtopic.php?f=159&t=31558#p230460

**Comment by JasonB32** [comment]
Thanks for trying @OtherCrashOverride , any emulators in the future are so exciting. I feel like a kid again with this product and can't wait for more emulators and features to come.

**Comment by OtherCrashOverride** [comment]
Atari 2600 is available here:
https://github.com/OtherCrashOverride/stella-odroid-go/releases

**Comment by JasonB32** [comment]
Amazing!!!! THANK YOU @OtherCrashOverride

**Comment by OtherCrashOverride** [comment]
Atari 7800 is available here:
https://github.com/OtherCrashOverride/prosystem-odroid-go/releases

**Comment by martin8777** [comment]
wow. this is awesome progress in such a short space of time. great work!

**Comment by JasonB32** [comment]
YES WOW, it's like Christmas in August. I couldn't believe it when I got home. I love these games so much, Thank you @OtherCrashOverride !!!

**Comment by JasonB32** [comment]
My Christmas Wishlist to Santa Claus:

1) MSX 2.
2) Commodore 64 (they have the best Arkanoid).
3) Amstrad - CPC.
4) Sharp X1.
5) Atari 5200 SuperSystem.
6) Intellivision (i know Colecovision is far superior already).

Other: Sierra Adventure Games Interpreter or Scumm VM (I don't know if this is even possible).

Bonus: Arduboy Emulator, Commander Keen series and Hugo's House of Horrors to be ported over hahaha!!!

The oDroid Go is so incredible I'm buying another one, once they come back in inventory their sold out right now.

Thank you again oDroid Go Team, and also thank you to Youtubers like Kamots Talk Tech and BackOfficeShow for educating me on how to flash the oDroid Go and teach me things about it.

**Comment by Schuemi** [comment]
Hi,
I am currently working on a fmsx port for the odroid go with the permission of Marat Fayzullin.

It looks very promising, Space Manbow for example works very well, I only have some sound problems I need to solve. When it's done, I'll upload it to github, maybe we can add it to this project here.

Edit: here is a small video showing Space Manbow. Som Video issues and no sound, but it runns smoothly :)

[https://www.youtube.com/watch?v=nZjO2yhPJHE](https://www.youtube.com/watch?v=nZjO2yhPJHE)

**Comment by JasonB32** [comment]
@Schuemi It's looking great, I'd love to have an MSX port. Of course sound would be really nice! Really looking forward to what you can do for this, thank you!

**Comment by OtherCrashOverride** [comment]
> maybe we can add it to this project here

It (fMSX) will need to have an open source license.  Currently modifications are prohibited without the author's permission.

**Comment by Schuemi** [comment]
Yes, you are right. Marat gave me the permission, but this definetly does not mean it's open source. I do the emulator as an independent project and with your cool flash firmware, it's no problem to change the emulators :)
